### PR TITLE
feat(SPV-1048): fixed metadata editing

### PR DESCRIPTION
**Pull request by Nazarii-4chain**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: engine/models_internal.go [hunk 1]
```diff
@@ -97,10 +97,26 @@ func (m *Model) SetRecordTime(created bool) {
 // UpdateMetadata will update the metadata on the model
 // any key set to nil will be removed, other keys updated or added
 func (m *Model) UpdateMetadata(metadata Metadata) {
```
Comment (chris-4chain) [inline engine/models_internal.go:99]: Why don't use a "normal" assigmnent?

A deep copy or shallow copy would only be necessary if the incoming metadata were modified elsewhere after being passed. However, in our case—as far as I can tell—there is no such scenario.

Please check if my solution works:

```golang
func (m *Model) UpdateMetadata(metadata Metadata) {
	m.Metadata = metadata
}
```

**Review by chris-4chain — CHANGES_REQUESTED** [review]

File: engine/models_internal.go (hunk 1 above)
Comment (Nazarii-4chain) [reply engine/models_internal.go:99]: It doesn't handle case of cleared all metadata when given an empty input 
I've got this error: `panic recovered: assignment to entry in nil map` If I want to change a metadata after deleting

Comment (chris-4chain) [reply engine/models_internal.go:99]: So? Just handle the empty-map case.

Comment (Nazarii-4chain) [reply engine/models_internal.go:99]: Updated

Comment (chris-4chain) [reply engine/models_internal.go:99]: I tested my solution on my own. It worked on my side.
When have you got the `panic recovered` error. 

Did you copy paste my code:
```golang
func (m *Model) UpdateMetadata(metadata Metadata) {
	m.Metadata = metadata
}
```
which you should - or you've added some lines below?

Comment (Nazarii-4chain) [reply engine/models_internal.go:99]: With line above. Added your solution

**Review by chris-4chain — APPROVED** [review]

**Review by jakubmkowalski — APPROVED** [review]

**Pull request merged**